src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,184 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> oldLast = last;
+        last = new Node<>(null, value, oldLast);
+        if (oldLast == null) {
+            first = last;
+        } else {
+            oldLast.next = last;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        outOfBoundsCheckToAdd(index);
+        Node<T> nodeToAdd = new Node<>(null, value, null);
+        Node<T> nodeNext = findNodeByIndex(index);
+        if (index == 0) {
+            first = nodeToAdd;
+            if (nodeNext != null) {
+                nodeToAdd.next = nodeNext;
+                nodeNext.prev = nodeToAdd;
+            } else {
+                last = first;
+            }
+            size++;
+        } else if (index == size) {
+            add(value);
+        } else {
+            nodeToAdd.prev = nodeNext.prev;
+            nodeToAdd.next = nodeNext;
+            Node<T> nodePrev = nodeNext.prev;
+            nodeNext.prev = nodeToAdd;
+            nodePrev.next = nodeToAdd;
+            size++;
+        }
```

Comment:
nested code is harder to read, let's simplify your solution to use less if/else
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        if (index == 0) {
            Node<T> newNode = new Node<>(null, value, first);
            first.prev = newNode;
            first = newNode;
        } else {
            Node<T> current = findNodeByIndex(index);
            Node<T> newNode = new Node<>(current.prev, value, current);
            current.prev.next = newNode;
            current.prev = newNode;
        }
        size++;
```